doc/man1/roarcatpassfh.1
.\" roarcatpassfh.1:

.TH "roarcatpassfh" "1" "July 2010" "RoarAudio" "System User's Manual: roarcatpassfh"

.SH NAME

roarcatpassfh \- Send audio data to RoarAudio sound server using PASSFH

.SH SYNOPSIS

roarcatpassfh [OPTIONS]...

.SH DESCRIPTION

Send audio data to RoarAudio sound server using the PASSFH protocol command.

.SH "OPTIONS"

.TP
\fB--server SERVER\fR
Set server hostname

.TP
\fB--rate   RATE\fR
Set sample rate

.TP
\fB--bits   BITS\fR
Set bits per sample

.TP
\fB--chans  CHANNELS\fR
Set number of channels

.TP
\fB--codec  CODEC\fR
Set the codec

.TP
\fB--help\fR
Show this help

.SH "SEE ALSO"
\fBroartips\fR(7),
\fBlibroar\fR(7),
\fBRoarAudio\fR(7).

.SH "HISTORY"

For history information see \fBRoarAudio\fR(7).

.\" ll


ChangeLog
v. 0.3beta7 - ?
	* Added support for audio profiles
	* Updated libroarrsound
	* Added RPlay protocol to roard

v. 0.3beta6 - Thu Jun 24 2010 20:47 CEST
	Prereleases: 0: Fri Jun 11 2010 10:47 CEST;
	             1: Sat Jun 12 2010 11:07 CEST;
	             2: Wed Jun 16 2010 12:52 CEST
	* Added something like select() for VIOs
	* Fixed a lot warnings
	* disabled cmd codecfilter by default
	* Fixed error with roarinterconnect on non-OSS systems
	* Fixed some problems with minimal builds
	* Some Copyright fixes
	* much better VIO support for connections
	* much better OSS emulation
	* Updated win32 port
	* Updated RSound support
	* fixed pos calc in case of PCM->PCM converting (pr1)
	* fixed libroarpulse (pr2)

v. 0.3beta5 - Sun May 02 2010 12:41 CEST
	* added tool roardtmf
	* Support to store server address in X11's root window propertys
	* some small fixes
	* wrote a lot more libroarpulse code, some async API clients work now
	* added compbins for esddsp, artsdsp, audiooss, padsp
	* splited libroarpulse and libroarpulse-simple
	* upgraded some code to use VIO
	* more updates for Debian
	* fixed init script for Debian and Debian like systems
	* fixed some mixer stuff, a lot things work better now
	* added support for notify beeps
	* updated drivers
	* updated ROAR_{ERR,WARN,INFO,DBG}()
	* added syslog support
	* updated test suit
	* done a lot work on libroaross
	* fixed segfault in ALSA plugin
	* Added support for OSS to roarinterconnect(1)
	* Added server side RSound Emulation
	* Added libroarrsound, client side RSound Emulation
	* Added ALSA driver
	* Added RSound driver
	* Added PortAudio driver

v. 0.3beta4 - Sat Feb 06 2010 26:02 CET
	* Applyed patches for Debian
	* updated roarify to load libs of correct version
	* updated list of supported versions (complibs)
	* added header with common units
	* added support for PulseAudio Simple protocol
	* added support to libroaross to emulate DMX4Linux (write only)
	* added DSTR driver to roard
	* updated librarys' soname
	* do not link OpenSSL by default (licence incompatible?)
	* support for dynamic lib loading, plugin interface
	* support to for registering DSTR types
	* moved WinSock things out of main vio.[ch]
	* support auto-tool --prefix*=*

v. 0.3beta3 - Fri Jan 22 2010 22:17 CET
	* added some aRts/NAS/YIFF/PulseAudio tools to compbins
	* added support for comp libs to roar-config
	* added support for uni-directional streams in roarinterconnect
	* write much better support for config, including codec config
	* reduced the TCP overhead produced by speex
	* added support for speex preprocessor
	* get names of complibs from table, do not guess them
	* a lot small fixes
	* moved most of the mixer stuff out of roard into libroardsp
	* added flags immutable and enhance
	* test for signal.h
	* Corrected (DSTR) HTTP Protocol support so servers without headerflush
	  should now work, too.
	* added protocol support for icy
	* Added version numbers to lib names
	* updated manpages
	* wrote OSS emulation
	* fixed test scripts

v. 0.3beta2 - Mon Oct 05 2009 19:48 CEST
	* Added prethru flag
	* Added support to read info about client's network node
	* Ported libroaresd, libroarartsc
	* Fixed some problems with EsounD emulation
	* added support for compatibility binarys
	* added poll() support to libroarsndio
	* reject streams of unsupported codec

v. 0.3beta1 - Sat Sep 26 2009 25:50 CEST
	* incremented nummber of streams per client fron 4 to 8
	* RIFF WAVE codecfilter uses virtual streams
	* another set of changed names of parameters in roarctl
	* Interface to read byteorder of a client from roard
	* implemented a 'state of stream' var
	* cleanup, small fixes
	* added VIO interface to transcoder interface
	* ask codecfilter of parent in case of new child stream
	* added 'subsystem' complex: multi subsystem data on single streams
	* added basic /Radio Data and Trasmitter Control System/.
	* better support for stereo upmixing
	* make audio group a optional thing
	* better support for win32
	* wrote WMM driver
	* better multi protocol support
	* Support for server side EsounD emulation on win32

v. 0.3beta0 - Sun Sep 06 2009 22:42 CEST
	* added new stream directions for MIDI and light control
	* cleanup of client stream struct
	* added basic light control subsystem
	* added DMX512 driver
	* wrote basic MIDI subsystem
	* added flag 'mute'
	* added flag 'mmap'
	* added lib libroarmidi and libroarlight
	* added OpenSLP support
	* rewote fillmixbuer2 code
	* added a roar_conv2() with a more clear API
	* added support to write pidfile
	* added support for encoding 32 bit Vorbis streams
	* added transcoding features in libroardsp
	* added tool 'roarphone'
	* added tool 'roarinterconnect'
	* added simple raw data subsystem
	* added working support for THRU streams
	* changed names of parameters in roarctl
	* added lib libroareio - Extended IO for RoarAudio
	* Update order of default drivers, now it is: oss, ao, esd, null
	* cleanup of old sysio driver interface
	* added server side EsounD emulation
	* added multi homed support to server
	* fixed a long outstanding bug with pa_path_get_filename()

v. 0.2 - Thu May 21 2009 21:00
	* do a some more testings for functions not existing on win32
	* init winsock
	* done a lot fixes in order to get libroar and some clients work on win32
	* done a lot cleanup on configure script

v. 0.2rc2 - Sun May 17 2009 25:44
	* corrected check for GNU peercred
	* added basic logging support to configure
	* fixed plugin configures
	* added and fixed semi-install target on plugins
	* call plugin configures from main configure
	* include plugins in Makefile known to work from configure
	* fixed Makefiles of plugins: added -I../../include, -L../../lib
	* Make use of pkg-config in plugin configures
	* corrected possition of -ls on linker commands in most Makefiles
	* added more explicide linker commands so most things now build
	  with classic BSD make.
	* devided into components: libs, roard, clients, doc

roarclients/roarcatpassfh.c
//roarcatpassfh.c:

/*
 *      Copyright (C) Philipp 'ph3-der-loewe' Schafft - 2008-2010
 *
 *  This file is part of roarclients a part of RoarAudio,
 *  a cross-platform sound system for both, home and professional use.
 *  See README for details.
 *
 *  This file is free software; you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License version 3
 *  as published by the Free Software Foundation.
 *
 *  RoarAudio is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License
 *  along with this software; see the file COPYING.  If not, write to
 *  the Free Software Foundation, 51 Franklin Street, Fifth Floor,
 *  Boston, MA 02110-1301, USA.
 *
 */

#include <roaraudio.h>

#define BUFSIZE (1024*2)

void usage (void) {
 printf("roarcatpassfh [OPTIONS]...\n");

 printf("\nOptions:\n\n");

 printf("  --server SERVER    - Set server hostname\n"
        "  --rate   RATE      - Set sample rate\n"
        "  --bits   BITS      - Set bits per sample\n"
        "  --chans  CHANNELS  - Set number of channels\n"
        "  --codec  CODEC     - Set the codec\n"
        "  --help             - Show this help\n"
       );

}

int main (int argc, char * argv[]) {
 int    rate     = 44100;
 int    bits     = 16;
 int    channels = 2;
 int    codec    = ROAR_CODEC_DEFAULT;
 char * server   = NULL;
 char * k;
 int    i;
 int    in = ROAR_STDIN;
 struct roar_connection con;
 struct roar_stream     s;


 for (i = 1; i < argc; i++) {
  k = argv[i];

  if ( strcmp(k, "--server") == 0 ) {
   server = argv[++i];
  } else if ( strcmp(k, "--rate") == 0 ) {
   rate = atoi(argv[++i]);
  } else if ( strcmp(k, "--bits") == 0 ) {
   bits = atoi(argv[++i]);
  } else if ( strcmp(k, "--channels") == 0 || strcmp(k, "--chans") == 0 ) {
   channels = atoi(argv[++i]);
  } else if ( strcmp(k, "--codec") == 0 ) {
   codec = roar_str2codec(argv[++i]);
  } else if ( strcmp(k, "--help") == 0 ) {
   usage();
   return 0;
  } else {
   fprintf(stderr, "Error: unknown argument: %s\n", k);
   usage();
   return 1;
  }
 }

 if ( roar_simple_connect(&con, server, "roarcatpassfh") == -1 ) {
  ROAR_DBG("roar_simple_play(*): roar_simple_connect() faild!");
  return 1;
 }

 if ( roar_stream_new(&s, rate, channels, bits, codec) == -1 ) {
  roar_disconnect(&con);
  return 1;
 }

 if ( roar_stream_connect(&con, &s, ROAR_DIR_PLAY) == -1 ) {
  roar_disconnect(&con);
  return 1;
 }

 if ( roar_stream_passfh(&con, &s, in) == -1 ) {
  roar_disconnect(&con);
  return 1;
 }

 close(in);

 sleep(1000);

 roar_disconnect(&con);

 return 0;
}

//ll
